File: app/src/main/java/com/kaltura/kflow/presentation/recordingList/RecordingListAdapter.java
package com.kaltura.kflow.presentation.recordingList;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import com.kaltura.client.types.Recording;
import com.kaltura.kflow.R;

import java.util.ArrayList;

/**
 * Created by alex_lytvynenko on 30.11.2018.
 */
public class RecordingListAdapter extends RecyclerView.Adapter<RecordingListAdapter.MyViewHolder> {
    private ArrayList<Recording> mRecordings = new ArrayList<>();
    private RecordingListAdapter.OnRecordingClickListener mListener;

    RecordingListAdapter(ArrayList<Recording> recordings, RecordingListAdapter.OnRecordingClickListener listener) {
        mRecordings.addAll(recordings);
        mListener = listener;
    }

    @NonNull
    @Override
    public RecordingListAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RecordingListAdapter.MyViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recording, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecordingListAdapter.MyViewHolder holder, int position) {
        holder.bind(mRecordings.get(position), mListener);
    }

    @Override
    public int getItemCount() {
        return mRecordings.size();
    }

    public interface OnRecordingClickListener {
        void onRecordingClicked(Recording recording);
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        private AppCompatTextView mStatus;
        private AppCompatTextView mId;
        private View mContainer;

        MyViewHolder(View v) {
            super(v);
            mContainer = v.findViewById(R.id.recording_container);
            mStatus = v.findViewById(R.id.recording_status);
            mId = v.findViewById(R.id.recording_id);
        }

        void bind(final Recording recording, final RecordingListAdapter.OnRecordingClickListener clickListener) {
            mStatus.setText(recording.getStatus().getValue());
            mId.setText("Asset ID: " + recording.getAssetId());
            mContainer.setOnClickListener(view -> clickListener.onRecordingClicked(recording));
        }
    }
}

File: app/src/main/java/com/kaltura/kflow/presentation/recordingList/RecordingListAdapter.kt
package com.kaltura.kflow.presentation.recordingList

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kaltura.client.types.Recording
import com.kaltura.kflow.databinding.ItemRecordingBinding
import kotlinx.android.extensions.LayoutContainer

/**
 * Created by alex_lytvynenko on 30.11.2018.
 */
class RecordingListAdapter : RecyclerView.Adapter<RecordingListAdapter.MyViewHolder>() {

    var recordingClickListener: (recording: Recording) -> Unit = {}

    var recordings: Array<Recording> = arrayOf()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val bindMe = ItemRecordingBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return MyViewHolder(bindMe,parent)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) = holder.bind(recordings[position])

    override fun getItemCount() = recordings.size
    private var _binding: ItemRecordingBinding? = null
    private val binding get() = _binding!!
    inner class MyViewHolder(val binding: ItemRecordingBinding,override val containerView: View) : RecyclerView.ViewHolder(binding.root), LayoutContainer {

        fun bind(recording: Recording) {
            binding.recordingStatus.text = recording.status.value
            binding.recordingId.text = "Asset ID: ${recording.assetId}"
            binding.recordingContainer.setOnClickListener { recordingClickListener(recording) }
        }
    }

//    inner class MyViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {
//
//        fun bind(recording: Recording) {
//            binding.recordingStatus.text = recording.status.value
//            binding.recordingId.text = "Asset ID: ${recording.assetId}"
//            binding.recordingContainer.setOnClickListener { recordingClickListener(recording) }
//        }
//    }
}